### PR TITLE
PST-406 allow monolog 2

### PR DESCRIPTION
**Pull request by pivnicek**
I think this should be enough

### CODE REVIEW

**Comment by pivnicek** [comment]
jo vyskousim ho v importeru

**Comment by pepamartinec** [comment]
Whoah, I just noticed the `LogProcessor` was not updated to accept the `LogRecord` at all :sweat_smile: 

`LogProcessor` should be changed in a following way:
```
public function(array|LogRecord $record): array
{
    if ($record instanceof LogRecord) {
        $record = $record->toArray():
    }

```

**Comment by odinuv** [comment]
Já bych se teda přikláněl k té variantě dát do test matrix PHP 8.0, kde monolog 3 nepojede, tak aby se to otestovalo i s monologem 2. S tím, že teda víme, že PHP8.0 nikde sice  nepoužíváme, ale potřebujeme monolog 2

**Comment by pivnicek** [comment]
jo, odin, to je proc tam mam 8.0, to jsou testy proti monolog 2

**Comment by pivnicek** [comment]
ok guys, it's green :-).  I hope I haven't missed anything

File: phpstan.neon [hunk 1]
```diff
@@ -5,6 +5,8 @@ parameters:
         - src
         - tests
     ignoreErrors:
+        #- '#Method Keboola\ErrorControl\Monolog\LogProcessor::__invoke() should return array{message: string, context: array, level: 100|200|250|300|400|500|550|600, level_name: 'ALERT'|'CRITICAL'|'DEBUG'|'EMERGENCY'|'ERROR'|'INFO'|'NOTICE'|'WARNING', channel: string, datetime: DateTimeImmutable, extra:array}|Monolog\LogRecord but returns non-empty-array.#'
+        #- '#Parameter #1 $record of callable Keboola\ErrorControl\Monolog\LogProcessor expects array{message: string, context: array, level: 100|200|250|300|400|500|550|600, level_name: 'ALERT'|'CRITICAL'|'DEBUG'|'EMERGENCY'|'ERROR'|'INFO'|'NOTICE'|'WARNING', channel: string, datetime: DateTimeImmutable, extra: array}|Monolog\LogRecord, array{(.*)} given.#'
```
Comment (pepamartinec) [inline phpstan.neon:9]: Remove these pls

File: src/Monolog/LogProcessor.php [hunk 2]
```diff
@@ -49,15 +67,11 @@ public function processRecord(array $record): array
                 'trace' => $exception->getTraceAsString(),
             ];
         }
+        $record['context'] = $context;
+        $record['extra']['app'] = $this->appName;
+        $record['extra']['pid'] = getmypid();
+        $record['extra']['priority'] = $record['level_name'];
```
Comment (pepamartinec) [inline src/Monolog/LogProcessor.php:73]: Najs

**Review by pepamartinec — APPROVED** [review]
Awesome! :heart:

File: src/Monolog/LogProcessor.php [hunk 3]
```diff
@@ -4,15 +4,29 @@
...
+ * @phpstan-type LevelName 'DEBUG'|'INFO'|'NOTICE'|'WARNING'|'ERROR'|'CRITICAL'|'ALERT'|'EMERGENCY'
+ * @phpstan-type Record array{
+ *      message: string,
+ *      context: mixed[],
+ *      level: Level,
+ *      level_name: LevelName,
+ *      channel: string,
+ *      datetime: \DateTimeImmutable,
+ *      extra: mixed[],
+ *      app?: string,
```
Comment (odinuv) [inline src/Monolog/LogProcessor.php:24]: tytok 3 řádky teda pryč když je to přesunuté do extra

**Review by odinuv — APPROVED** [review]
please fix Warning:       XDEBUG_MODE=coverage or xdebug.mode=coverage has to be set = that means just remove the coverage here https://github.com/keboola/api-error-control/blob/b2790746395b67afc99b8fc83e5171e1c2ed6219/composer.json#L39, it's not used for anything

**Pull request merged**